addons/website_mass_mailing/views/snippets_templates.xml

```diff
@@ -148,4 +136,21 @@
         </td>
     </xpath>
 </template>
+
+<template id="s_newsletter_popup_template" name="Newsletter">
+    <section class="s_mm_popup_content container">
+        <div class="row">
```

Comment:
There is a problem with this. If the user removes the `<section>`, he cannot add it back because this is not a snippet. I think the modal-body should be a `oe_structure` so that the user can drop any snippet in the modal. Then you can create a snippet for this, under the name "Snippet block". I think the code of the "subscribe" widget and the "popup" widget can be shared (both `_onClickSubscribe` looks similar). So I imagine something like this: the subscribe input is not added dynamically anymore. It has to be placed via a newletter / newletter block snippet inside the popup. This would allow to remove the duplicated JS code and will allow customizations to put the subscribe input anywhere in the modal. Here if the user deletes the last col or duplicates it or mess with the content a lot, the subscribe input will be added at a strange position.